src/core/net/ip6.cpp

```diff
@@ -1293,7 +1293,7 @@ Error Ip6::HandleDatagram(Message &aMessage, Netif *aNetif, const void *aLinkMes
 #endif
 
         // `SendMessage()` takes custody of message in the success case
-        SuccessOrExit(error = Get<ThreadNetif>().SendMessage(aMessage));
+        SuccessOrExit(error = Get<MeshForwarder>().SendMessage(aMessage));
```

Comment:
I think we can still use ThreadNetif for sending messages?

It's natural to say that messages are sent via the Thread network interface. 

Thoughts?

Reply:
I re-added it back in latest push. I agree it makes sense to want to send message via Thread netif  (at some point, we may combine `Netif and `ThreadNetif` into one class and we can re-consider this). Also it did not fit well with the main change in this PR.